plugin.video.akibapass/resources/lib/view.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# Akibapass - Watch videos from the german anime platform Akibapass.de on Kodi.
+# Copyright (C) 2016 - 2017 MrKrabat
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import sys
+import urllib
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+
+
+def endofdirectory():
+    # Sort methods are required in library mode
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_NONE)
+
+    # Let xbmc know the script is done adding items to the list
+    dontAddToHierarchy = False
+    xbmcplugin.endOfDirectory(handle = int(sys.argv[1]), updateListing = dontAddToHierarchy)
+
+
+def add_item(args, info, isFolder=True, total_items=0, queued=False, rex=re.compile(r"(?<=mode=)[^&]*"), mediatype="video"):
+    """Add item to directory listing.
+    """
+    # Defaults in dict. Use "None" instead of None so it is compatible for
+    info = set_info_defaults(args, info)
+    u = build_url(info)
+
+    # Create list item
+    li = xbmcgui.ListItem(label          = info["title"],
```

Comment:
Can we move this to setart? IconImage and thumbnailImage were deprecated jarvis and up :)
http://mirrors.xbmc.org/docs/python-docs/16.x-jarvis/xbmcgui.html#ListItem